azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/manager/AudioSessionManager.kt

```diff
@@ -32,37 +41,81 @@ internal class AudioSessionManager(
             }
             previousAudioDeviceSelectionStatus = it.localParticipantState.audioState.device
         }
+
+
+    }
+
+    override fun onAudioDevicesAdded(addedDevices: Array<out AudioDeviceInfo>?) {
+        super.onAudioDevicesAdded(addedDevices)
+        refreshDevices()
+    }
+
+    override fun onAudioDevicesRemoved(removedDevices: Array<out AudioDeviceInfo>?) {
+        super.onAudioDevicesRemoved(removedDevices)
+        refreshDevices()
+    }
+
+    private fun refreshDevices() {
+        audioDevices = audioManager.getDevices(AudioManager.GET_DEVICES_OUTPUTS)
```

Comment:
What will happen if on a device where API <= 22?

Reply:
21 should be the SDK limit here (our minSdk).

Reply:
actually, nvm. My IDE isn't complaining, but it's wrong, api 23 for registerAudioDeviceCallback (good catch).

I'll look for an alternative way to listen for this.

Reply:
going to resolve since these api's are all lowered now